src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -358,14 +358,19 @@ private JObject GenerateDependencies(JObject projectJsonRoot, ITaskItem[] extern
                 {
                     NuGetVersion nuGetVersion = NuGetVersion.Parse(dependency.GetMetadata("Version"));
                     PackageItem packageItem = new PackageItem(name, nuGetVersion);
+                    bool? forceVersion = dependency.GetMetadata("ForceVersion")?.Equals("true", StringComparison.OrdinalIgnoreCase);
```

Comment:
I'm not sure I fully understand what this is going to do. It isn't clicking in my head where ForceVersion will show-up and how it actually applies to the rewritting. @chcosta should also look at this change.